### PR TITLE
Mccalluc/test hi glass track component

### PR DESCRIPTION
**Pull request by mccalluc**
## Description

What was changed in this pull request?
- #693 didn't actually do what I needed it to do: The API was still not available when I needed it, and think because the render method on the inner component was not being called.
- This PR handles the composition of components in a more idiomatic way.
- To help with the test, I've tweaked `TiledPlot.waitForDomAttachment`... The live demo works, but `document.body.contains` does not work in the test... It's not really mounted in the DOM?

Why is it necessary?
- Still trying to get a react component that can show a 2d track and serve as a background for other tools.

## Checklist

- [ ] Unit tests added or updated
- [ ] Documentation added or updated
- [ ] Example added or updated
- [ ] Update schema.json if there are changes to the viewconf JSON structure format
- [ ] Screenshot for visual changes (e.g. new tracks)
- [ ] Updated CHANGELOG.md

### CODE REVIEW

File: app/scripts/HiGlassTrackComponent.js [hunk 1]
```diff
@@ -1,69 +1,54 @@
...
+            x: 0,
+            y: 0,
+            moved: false,
+            static: false,
+          },
+        },
+      ],
+    };
+    this.options = { bounded: true };
+    this.HgcRef = React.createRef();
```
Comment (pkerpedjiev) [inline app/scripts/HiGlassTrackComponent.js]:
```suggestion
    this.hgcRef = React.createRef();
```
camelCase for something that isn't a class, no?

File: app/scripts/HiGlassTrackComponent.js [hunk 2]
```diff
@@ -95,12 +69,18 @@ export default class HiGlassTrackComponent extends React.Component {
...
-    this.viewer.zoomTo(this.viewUid, x, x + width, y, y + height);
+    this.HgcRef.current.api.zoomTo(this.viewUid, x, x + width, y, y + height);
   }
 
   render() {
     return (
-      <div id={this.id} style={{ height: '100%', width: '100%' }} />
+      <div id={this.id} style={{ height: '100%', width: '100%' }}>
+        <HiGlassComponent
+          ref={this.HgcRef}
```
Comment (pkerpedjiev) [inline app/scripts/HiGlassTrackComponent.js]:
```suggestion
          ref={this.hgcRef}
```

File: app/scripts/HiGlassTrackComponent.js [hunk 3]
```diff
@@ -95,12 +69,18 @@ export default class HiGlassTrackComponent extends React.Component {
    * @param  {Number}  height  Height of viewport
    */
   zoomTo(x, y, width, height) {
-    this.viewer.zoomTo(this.viewUid, x, x + width, y, y + height);
+    this.HgcRef.current.api.zoomTo(this.viewUid, x, x + width, y, y + height);
```
Comment (pkerpedjiev) [inline app/scripts/HiGlassTrackComponent.js]:
```suggestion
    this.hgcRef.current.api.zoomTo(this.viewUid, x, x + width, y, y + height);
```

File: app/scripts/HiGlassTrackComponent.js [hunk 4]
```diff
@@ -95,12 +69,18 @@ export default class HiGlassTrackComponent extends React.Component {
...
   }
 
   render() {
     return (
-      <div id={this.id} style={{ height: '100%', width: '100%' }} />
+      <div id={this.id} style={{ height: '100%', width: '100%' }}>
```
Comment (pkerpedjiev) [inline app/scripts/HiGlassTrackComponent.js]: Why is this necessary? Can't you just return HiGlassComponent?

**Comment by flekschas** [comment]
Just as an FYI: here is how I initialize higlass from another react application in higlass-app and peax:

https://github.com/Novartis/peax/blob/develop/ui/src/components/HiGlassLauncher.js#L134-L145

And this is how cTracks does it. It's kind of similar: https://github.com/visdesignlab/cTracks/blob/master/src/node_modules/components/CTracksComp/HiglassUI.js#L89-L104

**Comment by mccalluc** [comment]
@manzt / @pkerpedjiev -- I'm closing out old PRs. I really can't remember what problem this was trying to solve. I'll leave the branch, though: if someone wants to start a new PR from it, you're free to do so.
